sunkit_image/granule.py

```diff
@@ -0,0 +1,297 @@
+    """
+    Get the threshold value using given skimage segmentation type.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Data to threshold.
+    method : {"li", "otsu", "isodata", "mean", "minimum", "yen", "triangle"}
+        scikit-image thresholding method.
+
+    Returns
+    -------
+    threshold : `float`
+        Threshold value.
+    """
+    if not isinstance(data, np.ndarray):
+        raise ValueError("Input data must be an instance of a np.ndarray")
+    if method == "li":
+        threshold = skimage.filters.threshold_li(data)
+    elif method == "otsu":
+        threshold = skimage.filters.threshold_otsu(data)
+    elif method == "yen":
+        threshold = skimage.filters.threshold_yen(data)
+    elif method == "mean":
+        threshold = skimage.filters.threshold_mean(data)
+    elif method == "minimum":
+        threshold = skimage.filters.threshold_minimum(data)
+    elif method == "triangle":
+        threshold = skimage.filters.threshold_triangle(data)
+    elif method == "isodata":
+        threshold = skimage.filters.threshold_isodata(data)
+    else:
+        raise ValueError("Method must be one of: " + ", ".join(METHODS))
+    return threshold
+
+
+def trim_intergranules(segmented_image, mark=False):
+    """
+    Remove the erroneous identification of intergranule material in the middle
+    of granules that the pure threshold segmentation produces.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect extra intergranules.
+    mark : `bool`
+        If `False` (the default), remove erroneous intergranules.
+        If `True`, mark them as 0.5 instead (for later examination).
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndarray`
+        The segmented image without incorrect extra intergranules.
+    """
+    if len(np.unique(segmented_image)) > 2:
+        raise ValueError("segmented_image must only have values of 1 and 0.")
+    # Float conversion for correct region labeling.
+    segmented_image_fixed = np.copy(segmented_image).astype(float)
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    # Find value of the large continuous 0-valued region.
+    size = 0
+    for value in values:
+        if len((labeled_seg[labeled_seg == value])) > size:
+            real_IG_value = value
+            size = len(labeled_seg[labeled_seg == value])
+    # Set all other 0 regions to mark value (1 or 0.5).
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+    return segmented_image_fixed
+
+
+def mark_brightpoint(segmented_image, data, resolution, bp_min_flux):

```

Comment:
```suggestion
def mark_brightpoint(segmented_image, data, resolution, bp_min_flux=None):
```

I think it might be nice to default this to None.